### PR TITLE
Automatically update copyright years of all sources

### PR DESCRIPTION
**Pull request by ibuclaw**
These sources were rejected, and updated by hand.
```
./src/dmd/backend/cdef.d(365): unrecognised copyright: //#define COPYRIGHT "Copyright " COPYRIGHT_SYMBOL " 2001 The D Language Foundation"
./src/dmd/backend/cdef.d(368): unrecognised copyright: //#define COPYRIGHT "Copyright (C) The D Language Foundation 2000-2017.  All Rights Reserved.\n\
./src/dmd/backend/cdef.d(373): unrecognised copyright: //#define COPYRIGHT "Copyright (C) The D Language Foundation 2000-2017.  All Rights Reserved.\n\
./src/dmd/backend/cdef.d(376): unrecognised copyright: //#define COPYRIGHT "Copyright (C) The D Language Foundation 2000-2017.  All Rights Reserved.\n\
./src/dmd/backend/cdef.h(602): unrecognised copyright: #define COPYRIGHT "Copyright " COPYRIGHT_SYMBOL " 2001-2017 The D Language Foundation"
./src/dmd/backend/cdef.h(605): unrecognised copyright: #define COPYRIGHT "Copyright (C) The D Language Foundation 2000-2017.  All Rights Reserved.\n\
./src/dmd/backend/cdef.h(610): unrecognised copyright: #define COPYRIGHT "Copyright (C) The D Language Foundation 2000-2017.  All Rights Reserved.\n\
./src/dmd/backend/cdef.h(613): unrecognised copyright: #define COPYRIGHT "Copyright (C) The D Language Foundation Mars 2000-2017.  All Rights Reserved.\n\
```

All automated editing was done by https://github.com/dlang/tools/pull/261

### CODE REVIEW

File: src/dmd/backend/blockopt.c [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * $(LINK2 http://www.dlang.org, D programming language).
  *
  * Copyright:   Copyright (C) 1986-1997 by Symantec
- *              Copyright (c) 2000-2012 by The D Language Foundation, All Rights Reserved
```
Comment (wilzbach) [inline src/dmd/backend/blockopt.c]: :scream: - it's good that we automate this :)

**Review by wilzbach — APPROVED** [review]
Splitting it up in separate commits was very helpful. Thanks!
Also great work with the script - looks like its doing its job perfectly :)

**Comment by ibuclaw** [comment]
Removed auto-merge, I spotted some missed cases not handled by automation. :-(

i.e: double `The`.
```
./src/dmd/asttypename.d(7): unrecognised copyright holder: The The D Language Foundation, All Rights Reserved
```

**Comment by ibuclaw** [comment]
OK, these are the remaining copyright failures after fixing a small logic bug. ;-)
```
./src/dmd/asttypename.d(4): unrecognised copyright holder: The The D Language Foundation, All Rights Reserved
./src/dmd/backend/cdef.d(365): unrecognised copyright: //#define COPYRIGHT "Copyright " COPYRIGHT_SYMBOL " 2001-2018 by The D Language Foundation"
./src/dmd/backend/cdef.h(602): unrecognised copyright: #define COPYRIGHT "Copyright " COPYRIGHT_SYMBOL " 2001-2018 by The D Language Foundation"
```

**Comment by ibuclaw** [comment]
@wilzbach - Sorry bout that, it's all good now!  (It even updates the help string).

**Comment by wilzbach** [comment]
If you want to go for full completeness you might want to bump the [`dub.sdl`](https://github.com/dlang/dmd/blob/master/dub.sdl) file too.
The man page will hopefully be generated automatically [soon](https://github.com/dlang/dmd/pull/7507).

**Comment by ibuclaw** [comment]
> If you want to go for full completeness you might want to bump the dub.sdl file too.

Manual or automated? :-P

**Comment by wilzbach** [comment]
> Manual or automated? :-P

I quickly did it manually for this year.

**Pull request merged**